kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/Serialization.java

```diff
@@ -253,17 +253,32 @@ public Type getType() {
   }
 
   private static boolean containsMultipleDocuments(String specFile) {
-    Pattern p = Pattern.compile(DOCUMENT_DELIMITER);
-    Matcher m = p.matcher(specFile);
-    int count = 0;
-    while (m.find()) {
-      count++;
+    String[] documents = splitSpecFile(specFile);
+    int nValidDocuments = 0;
+    for(String document : documents) {
+      if(validate(document))
+        nValidDocuments++;
     }
-    if (count == 1) {
-      String[] documents = specFile.split(DOCUMENT_DELIMITER);
-      return validate(documents[0]);
-    }
-    return count > 1;
+
+    return nValidDocuments > 1;
+  }
+
+  private static String[] splitSpecFile(String aSpecFile) {
+    List<String> documents = new ArrayList<>();
+    String[] lines = aSpecFile.split(System.lineSeparator());
+    int nLine = 0;
+
+    do {
```

Comment:
This thing can be reduced to a single loop in place of a loop in the loop. This is just decreasing the code readability. You can just use a single loop which will iterate over all the line and whenever you see a `DOCUMENT_DELIMETER` add the previous string to documents List and start a new string. The complexity of both is same but why to increase the reading complexity if we can achieve it in a single loop.